api/v1alpha7/openstackmachine_types.go

```diff
@@ -84,6 +84,9 @@ type OpenStackMachineSpec struct {
 	// The volume metadata to boot from
 	RootVolume *RootVolume `json:"rootVolume,omitempty"`
 
+	// DataVolumes is a list of specifications for volumes to attach to the server instance
+	DataVolumes []DataVolume `json:"dataVolumes,omitempty"`
```

Comment:
Not a big deal, but just to mention the CAPZ used `DataDisks`.
I wonder if we want to follow that pattern, since our data might end-up being on volumes and/or local disks at some point.

Reply:
I used `DataVolumes` to match `RootVolume`. I don't really have a strong opinion.

Reply:
The things you have to set in kubeadmcontrolplane and/or kubeadmbootstrapconfig correspond to cloud-init fields, so `DiskSetup` and `Mounts`.

Reply:
But the OpenStack terminology for these things is volume, so my feeling is that we stick with that 🤷‍♂️

Reply:
Could we settle on the API, considering that we'll support multiple ephemeral disks and volumes.

_Option A_. `DataVolumes` and `LocalDataVolumes` where each of them have the options needed for their use-case.

_Option B_. One field (maybe name it `BlockDevices` to avoid confusion with volumes`) with all fields that we need for both use cases.


I have a preference for _Option B_. (and the rename) for simplicity's sake.

Reply:
My preference is also for option B. I like using a name that mirrors the OpenStack API attribute that we are ultimately setting - I think it is probably less confusing for users.